### PR TITLE
Update entity class names for HA 0.110

### PR DESCRIPTION
**Pull request by mijost**
Adapt to changes described on https://developers.home-assistant.io/blog/2020/05/14/entity-class-names/

### CODE REVIEW

File: custom_components/apple_tv/media_player.py [hunk 1]
```diff
@@ -3,7 +3,10 @@
 
 from pyatv.const import DeviceState, FeatureName, FeatureState, MediaType
 
-from homeassistant.components.media_player import MediaPlayerDevice
+try:
+    from homeassistant.components.media_player import MediaPlayerEntity
+except ImportError:
+    from homeassistant.components.media_player import MediaPlayerDevice as MediaPlayerEntity
```
Comment (postlund) [inline custom_components/apple_tv/media_player.py:9]: No need to make it backwardscompatible, just import `MediaPlayerEntity`.

**Comment by postlund** [comment]
An oh, state that 0.110 is required so that people know that.

**Comment by postlund** [comment]
Just pushed update for this!

**Comment by zibous** [comment]
I use Apple TV via integrations, 

Bildschirmfoto 2020-06-11 um 09 22 14

but I always get the error message when I start:

```text
MediaPlayerDevice is deprecated, modify AppleTvDevice to extend MediaPlayerEntity
RemoteDevice is deprecated, modify AppleTVRemote to extend RemoteEntity
```
**Systemzustand Home Assistant 0.111.0**
```txt
arch | x86_64
-- | --
dev | false
docker | false
hassio | false
installation_type | Home Assistant Core
os_name | Linux
os_version | 5.4.0-37-generic
python_version | 3.8.2
timezone | Europe/Vaduz
version | 0.111.0
virtualenv | true
```
Bildschirmfoto 2020-06-11 um 09 22 30

How can I update Apple TV via integrations?

**Comment by postlund** [comment]
It depends on how you installed the component. I recommend that you use HACS as you get an easy way to upgrade. Just add this repo as a custom repository and install.

**Comment by zibous** [comment]
@postlund 
Thanks. I will try the installation with HACS. Currently i have used the Integration to install the component.
It looks like that the integration do not update the component.

HACS can i only use, when i add the 'Beta version of new Apple TV integration in Home Assistant supporting tvOS 13' as custom repository ?

**Comment by postlund** [comment]
When you say "Integration", what are you referring to? You have to install the component itself (the code) by copying it to `custom_components` and manually update the files when an update is available. You then add new devices from the integrations page. These are two different things.

If you use HACS, it can update the component automatically, but it requires (as you say) that you add this repository as a custom repository as it is not in the default pool of repositories.

**Comment by zibous** [comment]
hi @postlund 
> When you say "Integration", what are you referring to?
I used the Integration from the ui, but this do not load the latest version.

Bildschirmfoto 2020-06-11 um 10 21 57

Now i use HACS and i got the latest version.
Thanks 👍

**Comment by postlund** [comment]
Awesome! 👍

**Comment by zibous** [comment]
@postlund 
When I start apple tv and watch a movie, the status always remains on "idle".

Bildschirmfoto 2020-06-11 um 17 32 33

Is it because I have an old device "Apple TV (3rd generation Rev A)" or am I doing something wrong in the settings?

Bildschirmfoto 2020-06-11 um 17 37 01

**Comment by postlund** [comment]
It should work with ATV 3 as well. Can you try `atvremote --debug -s <ip> push_updates`, try playing something and see if it updates? Please attach any log as well.

**Comment by zibous** [comment]
@postlund 
Yes it is working:
```txt
atvremote --debug -s 10.1.1.167 push_updates
2020-06-11 18:16:41 DEBUG: Knocking at port 3689 on 10.1.1.167
2020-06-11 18:16:41 DEBUG: Knocking at port 7000 on 10.1.1.167
2020-06-11 18:16:41 DEBUG: Knocking at port 49152 on 10.1.1.167
2020-06-11 18:16:41 DEBUG: Knocking at port 32498 on 10.1.1.167
2020-06-11 18:16:41 DEBUG: Sending DNS request to 10.1.1.167 (Data=35ff012000040000000000000b5f6170706c6574762d7632045f746370056c6f63616c0000ff80010b5f746f7563682d61626c65045f746370056c6f63616c0000ff80010e5f6d6564696172656d6f74657476045f746370056c6f63616c0000ff8001085f616972706c6179045f746370056c6f63616c0000ff8001)
2020-06-11 18:16:41 DEBUG: Received DNS response from 10.1.1.167 (Data=35ff840000030003000000090b5f6170706c6574762d7632045f746370056c6f63616c0000ff80010b5f746f7563682d61626c65c01800ff8001085f616972706c6179c01800ff8001c03a000c00010000000a000e0b6d797476626f7820283229c03ac028000c00010000000a00131045414645334432463644354543374236c028c00c000c00010000000a00131045414645334432463644354543374236c00cc055002100010000000a0012000000001b58096d797476626f782d32c01dc055001000010000000a00de1a64657669636569643d36303a30333a30383a44333a37323a42371866656174757265733d307835413746464646372c3078314...)
2020-06-11 18:16:41 DEBUG: Cleaning up after UDNS request
2020-06-11 18:16:41 DEBUG: Cleaning up after UDNS request
2020-06-11 18:16:41 DEBUG: Auto-discovered mytvbox (2) at 10.1.1.167:7000 (Protocol.AirPlay)
2020-06-11 18:16:41 DEBUG: Auto-discovered mytvbox at 10.1.1.167:3689 (Protocol.DMAP)
2020-06-11 18:16:41 DEBUG: Auto-discovered mytvbox at 10.1.1.167:3689 (Protocol.DMAP)
2020-06-11 18:16:41 INFO: Auto-discovered mytvbox (2) at 10.1.1.167
2020-06-11 18:16:41 DEBUG: No AirPlay credentials loaded
2020-06-11 18:16:41 DEBUG: GET URL: http://10.1.1.167:3689/login?hsgid=00000000-4efe-ae9e-b917-192870134abc&hasFP=1
2020-06-11 18:16:41 DEBUG: Response: status=200, headers=[Date=Thu, 11 Jun 2020 16:16:41 GMT, DAAP-Server=iTunes/11.1b37 (OS X), Content-Type=application/x-dmap-tagged, Content-Length=47, Content-Encoding=gzip]
2020-06-11 18:16:41 DEBUG: << GET (Data=6d6c6f67000000186d73747400000004000000c86d6c6964000000047e848c9a)
2020-06-11 18:16:41 DEBUG: mlog: [container, dmap.loginresponse]
  mstt: 200 [uint, dmap.status]
  mlid: 2122615962 [uint, dmap.sessionid]

2020-06-11 18:16:41 INFO: Logged in and got session id 2122615962
Press ENTER to stop
2020-06-11 18:16:41 DEBUG: Waiting for playstatus updates
2020-06-11 18:16:41 DEBUG: Already logged in, re-using seasion id 2122615962
2020-06-11 18:16:41 DEBUG: GET URL: http://10.1.1.167:3689/ctrl-int/1/playstatusupdate?session-id=2122615962&revision-number=0
2020-06-11 18:16:41 DEBUG: Response: status=200, headers=[Date=Thu, 11 Jun 2020 16:16:41 GMT, DAAP-Server=iTunes/11.1b37 (OS X), Content-Type=application/x-dmap-tagged, Content-Length=41, Content-Encoding=gzip]
2020-06-11 18:16:41 DEBUG: << GET (Data=636d7374000000186d73747400000004000000c8636d73720000000400000045)
2020-06-11 18:16:41 DEBUG: cmst: [container, dmcp.playstatus]
  mstt: 200 [uint, dmap.status]
  cmsr: 69 [uint, dmcp.serverrevision]

2020-06-11 18:16:41 DEBUG: Waiting for playstatus updates
2020-06-11 18:16:41 DEBUG: Already logged in, re-using seasion id 2122615962
2020-06-11 18:16:41 DEBUG: GET URL: http://10.1.1.167:3689/ctrl-int/1/playstatusupdate?session-id=2122615962&revision-number=69
  Media type: Unknown
Device state: Idle
      Repeat: Off
     Shuffle: Off
--------------------
2020-06-11 18:16:50 DEBUG: Response: status=200, headers=[Date=Thu, 11 Jun 2020 16:16:50 GMT, DAAP-Server=iTunes/11.1b37 (OS X), Content-Type=application/x-dmap-tagged, Content-Length=254, Content-Encoding=gzip]
2020-06-11 18:16:50 DEBUG: << GET (Data=636d73740000014e6d73747400000004000000c8636d7372000000040000004663617073000000010163617368000000010063617270000000010063616673000000010063617673000000010063617663000000010063616173000000040000000263616172000000040000000663616665000000010063617665000000010063655141000000010063616e6e000000315343485745494e53544549474552204d656d6f72696573202d20566f6e20416e66616e6720626973204c6567656e646500636553440000006562706c6973743030d101025269645f1031616d7a6e312e64762e6774692e32306239333865332d303361392d343431622d3962633...)
2020-06-11 18:16:50 DEBUG: cmst: [container, dmcp.playstatus]
  mstt: 200 [uint, dmap.status]
  cmsr: 70 [uint, dmcp.serverrevision]
  caps: 1 [uint, dacp.playstatus]
  cash: 0 [uint, dacp.shufflestate]
  carp: 0 [uint, dacp.repeatstate]
  cafs: 0 [uint, dacp.fullscreen]
  cavs: 0 [uint, dacp.visualizer]
  cavc: False [bool, dacp.volumecontrollable]
  caas: 2 [uint, dacp.albumshuffle]
  caar: 6 [uint, dacp.albumrepeat]
  cafe: False [bool, dacp.fullscreenenabled]
  cave: False [bool, dacp.dacpvisualizerenabled]
  ceQA: 0 [uint, unknown tag]
  cann: SCHWEINSTEIGER Memories - Von Anfang bis Legende [str, daap.nowplayingtrack]
  ceSD: {'id': 'amzn1.dv.gti.20b938e3-03a9-441b-9bc1-1df489856c11'} [bplist, playing metadata]
  cmmk: 2 [uint, dmcp.mediakind]
  casc: 1 [uint, unknown tag]
  caks: 6 [uint, unknown tag]
  casu: 0 [uint, dacp.su]

2020-06-11 18:16:50 DEBUG: Waiting for playstatus updates
2020-06-11 18:16:50 DEBUG: Already logged in, re-using seasion id 2122615962
2020-06-11 18:16:50 DEBUG: GET URL: http://10.1.1.167:3689/ctrl-int/1/playstatusupdate?session-id=2122615962&revision-number=70
  Media type: Music
Device state: Loading
       Title: SCHWEINSTEIGER Memories - Von Anfang bis Legende
      Repeat: Off
     Shuffle: Off
--------------------
2020-06-11 18:16:52 DEBUG: Response: status=200, headers=[Date=Thu, 11 Jun 2020 16:16:52 GMT, DAAP-Server=iTunes/11.1b37 (OS X), Content-Type=application/x-dmap-tagged, Content-Length=267, Content-Encoding=gzip]
2020-06-11 18:16:52 DEBUG: << GET (Data=636d7374000001666d73747400000004000000c8636d7372000000040000004763617073000000010463617368000000010063617270000000010063616673000000010063617673000000010063617663000000010063616173000000040000000263616172000000040000000663616665000000010063617665000000010063655141000000010063616e6e000000315343485745494e53544549474552204d656d6f72696573202d20566f6e20416e66616e6720626973204c6567656e646500636553440000006562706c6973743030d101025269645f1031616d7a6e312e64762e6774692e32306239333865332d303361392d343431622d3962633...)
2020-06-11 18:16:52 DEBUG: cmst: [container, dmcp.playstatus]
  mstt: 200 [uint, dmap.status]
  cmsr: 71 [uint, dmcp.serverrevision]
  caps: 4 [uint, dacp.playstatus]
  cash: 0 [uint, dacp.shufflestate]
  carp: 0 [uint, dacp.repeatstate]
  cafs: 0 [uint, dacp.fullscreen]
  cavs: 0 [uint, dacp.visualizer]
  cavc: False [bool, dacp.volumecontrollable]
  caas: 2 [uint, dacp.albumshuffle]
  caar: 6 [uint, dacp.albumrepeat]
  cafe: False [bool, dacp.fullscreenenabled]
  cave: False [bool, dacp.dacpvisualizerenabled]
  ceQA: 0 [uint, unknown tag]
  cann: SCHWEINSTEIGER Memories - Von Anfang bis Legende [str, daap.nowplayingtrack]
  ceSD: {'id': 'amzn1.dv.gti.20b938e3-03a9-441b-9bc1-1df489856c11'} [bplist, playing metadata]
  cmmk: 2 [uint, dmcp.mediakind]
  casc: 1 [uint, unknown tag]
  caks: 6 [uint, unknown tag]
  cant: 115839 [uint, dacp.remainingtime]
  cast: 116000 [uint, dacp.tracklength]
  casu: 0 [uint, dacp.su]

2020-06-11 18:16:52 DEBUG: Waiting for playstatus updates
2020-06-11 18:16:52 DEBUG: Already logged in, re-using seasion id 2122615962
2020-06-11 18:16:52 DEBUG: GET URL: http://10.1.1.167:3689/ctrl-int/1/playstatusupdate?session-id=2122615962&revision-number=71
  Media type: Music
Device state: Playing
       Title: SCHWEINSTEIGER Memories - Von Anfang bis Legende
    Position: 0/116s (0.0%)
      Repeat: Off
     Shuffle: Off
--------------------
2020-06-11 18:17:00 DEBUG: Response: status=200, headers=[Date=Thu, 11 Jun 2020 16:17:00 GMT, DAAP-Server=iTunes/11.1b37 (OS X), Content-Type=application/x-dmap-tagged, Content-Length=41, Content-Encoding=gzip]
2020-06-11 18:17:00 DEBUG: << GET (Data=636d7374000000186d73747400000004000000c8636d73720000000400000048)
2020-06-11 18:17:00 DEBUG: cmst: [container, dmcp.playstatus]
  mstt: 200 [uint, dmap.status]
  cmsr: 72 [uint, dmcp.serverrevision]

2020-06-11 18:17:00 DEBUG: Waiting for playstatus updates
2020-06-11 18:17:00 DEBUG: Already logged in, re-using seasion id 2122615962
2020-06-11 18:17:00 DEBUG: GET URL: http://10.1.1.167:3689/ctrl-int/1/playstatusupdate?session-id=2122615962&revision-number=72
  Media type: Unknown
Device state: Idle
      Repeat: Off
     Shuffle: Off
--------------------
```
But the state HA GUI is every time "idle" - media_title: Nothing playing.

**Comment by postlund** [comment]
Ok, good. Then we at least know that it works with pyatv standalone. Now it would be interesting to see logs from Home Assistant as well. I suggest these log points:

```yaml

logger:
  default: critical
  logs:
    pyatv: debug
    custom_components.apple_tv: debug
```

**Comment by zibous** [comment]
@postlund 

Sorry, I don't understand. But after the restart it works now, even if I turn logging off again.

Bildschirmfoto 2020-06-11 um 18 41 55

Must watch it, but thanks for the help.

**Comment by postlund** [comment]
That is what you add to `configuration.yaml` to more or less disable all logging and enabling logging in Apple TV component as well as in pyatv. If you run into the problem again, please give that a go and see if it works.

**Pull request closed without merging**